src/lib/viz/layer/Layer.ts

```diff
@@ -471,25 +464,22 @@ export class Layer extends WithEvents implements StyledLayer {
     return Promise.resolve();
   }
 
+  addSourceField(field: string) {
+    this._source.addField(field);
```

Comment:
Should we check if the field already exists?

Reply:
The logic about reinitialized is managed by Source class where we have:
```
addField(field: string) {
  const { size } = this.fields;

  this.fields.add(field);

  if (size < this.fields.size) {
    this.needsInitialization = true;
  }
}
```

So the fields are a Set in order to avoid duplicates and in case a new element is added, `needsInitialization` is set to `true` and in the following `replaceDeckGLLayer` the source will be reinitialized

Reply:
Perfect, I didn't notice the Set.